### PR TITLE
modify the return value, to make it quit when the service controller  started failed

### PR DESCRIPTION
**Pull request by yan234280533**
**What this PR does / why we need it**:
Service controller is the important controller for kubernetes. If it startd failed , I think the kubernetes can not work normal in most senses. So I think it's an suitable way to make the controller manager explicitly show this error.

I modified the return value when the service controller  started failed. By return an error message, it make the main thread of  controller manager know.

### CODE REVIEW

File: cmd/kube-controller-manager/app/core.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func startServiceController(ctx ControllerContext) (bool, error) {
 	)
 	if err != nil {
 		glog.Errorf("Failed to start service controller: %v", err)
```
Comment (CaoShuFeng) [inline cmd/kube-controller-manager/app/core.go]: I think these startXXXController() functions should be consistent with each other and return false when started failed.

**Comment by CaoShuFeng** [comment]
> By return an error message, it make the main thread of controller manager know.

By returning `false nil` the main thread of controller also know that it started failed, because `false` is returned.

File: cmd/kube-controller-manager/app/core.go (hunk 1 above)
Comment (yan234280533) [reply cmd/kube-controller-manager/app/core.go]: @CaoShuFeng  Many thanks for you to help me to review this code . Did you mean that the startXXXController() functions return false, when it start failed ?

**Comment by yan234280533** [comment]
By returning false nil the main thread of controller also know that it started failed, because false is returned.

If it return false, the main thread will continue to start. So the kube-controller-manager will started, but it‘s major feature cannot running.

File: cmd/kube-controller-manager/app/core.go (hunk 1 above)
Comment (CaoShuFeng) [reply cmd/kube-controller-manager/app/core.go]: All [these](https://github.com/kubernetes/kubernetes/blob/ecd910ea6cb5158ac8543018a52a889e21cf3db2/cmd/kube-controller-manager/app/controllermanager.go#L315) functions should be consistent with each other and return the same value when start failed, right?

Comment (CaoShuFeng) [reply cmd/kube-controller-manager/app/core.go]: I found that in other places, also returned "true, err".
It's quite strange for me. I thought that `true` means started successfully.
My pervious comment is invalid, sorry!
Your return value is reasonable!

Comment (CaoShuFeng) [reply cmd/kube-controller-manager/app/core.go]: This is what deads2k says about the first return value, I'd like to share it with you:
aaa

Comment (yan234280533) [reply cmd/kube-controller-manager/app/core.go]: I think it right rule , "true, err" it mean I want to start it , but it failed.

Comment (yan234280533) [reply cmd/kube-controller-manager/app/core.go]: @deads2k  deads, the two return values, is it means:  the first it to show if we want to enable/start it , and the second is started it ok ?

Comment (deads2k) [reply cmd/kube-controller-manager/app/core.go]: > @deads2k deads, the two return values, is it means: the first it to show if we want to enable/start it , and the second is started it ok ?

Yes,but during the transition to init functions, we had some pre-existing controllers that failed silently.  I think this change is correct, but I'm not certain whether something will break on it.

Comment (yan234280533) [reply cmd/kube-controller-manager/app/core.go]: I think some controller is will start failed in some scenario.

As I do the CI , it failed.

I found that :+1: 

W0828 07:28:02.599] E0828 07:28:02.598300   13161 core.go:69] Failed to start service controller: WARNING: no cloud provider provided, services of type LoadBalancer will fail.
W0828 07:28:02.599] E0828 07:28:02.598319   13161 controllermanager.go:453] Error starting "service"
W0828 07:28:02.599] F0828 07:28:02.598325   13161 controllermanager.go:188] error starting controllers: WARNING: no cloud provider provided, services of type LoadBalancer will fail.
W0828 07:28:32.831] !!! [0828 07:28:32] Timed out waiting for controller-manager to answer at http://127.0.0.1:10252/healthz; tried 30 waiting 1 between each

From this message, in the  scenario if it not provided an cloud provider, the service-controller will started failed, but the controller-manager will start also.

**Pull request closed without merging**